torchgen/model.py

```diff
@@ -354,6 +356,14 @@ class NativeFunction:
     # defined.  This is for conveniently reporting error messages!
     loc: "Location"
 
+    # A list of operators that are expected to be auto-generated for this NativeFunction.
+    # Note: This list isn't actually directly used by the codegen to generate anything.
+    # Instead, the codegen figures out what operators to generate purely based off of
+    # function schema, and uses the autogen declarations to error check.
+    # We expect every NativeFunction that gets auto-generated be explicitly called out
+    # in native_functions.yaml
+    autogen: List["OperatorName"]
```

Comment:
Is this going to be enough, if we ever generate TWO operators from one function? It might be tricky to disambiguate which variant the operators are supposed to be, in that case.

Reply:
>  if we ever generate TWO operators from one function? 

This PR already does that today, so maybe I'm misunderstanding? E.g. for all of the `_foreach` ops, this codegen will generate  a `.functional` and a `.out` variant.

I'm also still under the assumption that we should be able to figure out what `NativeFunctions` to generate in the future entirely based off of schema parsing + grouping logic, and the list of op names here is really just for error checking + being explicit